Update Contributing Guide (#423)
holoviz/holoviz: doc/contribute.md

```{note}

> The HoloViz projects usually provides specific instructions on how to set up your environment to be ready to contribute documentation or code to the project. The instructions provided hereafter are meant to be general enough to apply to each project. Refer to the documentation of the project you are working on for more details.

Reviewer: ```suggestion
The HoloViz projects usually provide specific instructions on how to set up your environment to be ready to contribute documentation or code to the project. The instructions provided hereafter are meant to be general enough to apply to each project. Refer to the documentation of the project you are working on for more details.
```

    * [jupyter-panel-proxy](https://github.com/holoviz/jupyter-panel-proxy): Jupyter Server Proxy for Panel
* [holoviz-dev](https://github.com/holoviz-dev/)
    * [nbsite](https://github.com/holoviz/nbsite): Build a tested, sphinx-based website from notebooks

>     * [nbsmoke](https://github.com/holoviz-dev/nbsmoke): Basic notebook checks. Do they run? Do they contain lint?

Reviewer: We actually still use `nbsmoke` on this very own repo :D

Reviewer 2: OK. So I'll re-add it then

* The core maintainers of the HoloViz group are employed by [Anaconda](https://www.anaconda.com/), and as such, it makes sense for the group to use `conda`
* Some HoloViz-maintained packages such as `Panel` require software like `node.js`, which `pip` cannot install. In contrast, they can be installed with `conda`. `conda` allows to create a complete dev environment without having to install other software from another way, e.g., with `brew` on macOS.

> To contribute to HoloViz, we **recommend** that you use the `conda ecosystem`. You would have a closer experience with the maintainers, and they will be in a better position to help you set up your dev environment and debug whatever issue you may encounter.

Reviewer: ```suggestion
To contribute to HoloViz, we **recommend** that you use the *conda ecosystem*. You would have a closer experience with the maintainers, and they will be in a better position to help you set up your dev environment and debug whatever issue you may encounter.
```

#### Pixi

[Pixi](https://pixi.sh/latest/) is a fast software package manager built on top of the conda ecosystem. It is used to simplify the management of the environments and tasks across projects.

In each core HoloViz repository, you will find a few files that are of importance for `pixi`:

> * `pixi.toml` is where the environments and tasks is declared.

Reviewer: ```suggestion
* `pixi.toml` is where the environments and tasks are declared.
```

> * `pixi.lock` is the lock file that is generated by `pixi` and that contains the exact versions of the dependencies used in the environment.

Reviewer: ```suggestion
* `pixi.lock` (ignored by `git`)  is the lock file that is automatically generated by `pixi` when running commands that need to create an environment; it contains the exact versions of the dependencies used in the environment.
```

> * `scripts` is a directory that contains scripts that which is used by `pixi`.

Reviewer: ```suggestion
* `scripts` is a directory that contains scripts (Python, bash, etc.) called by some `pixi` tasks.
```

Some of the sites have their own domain name:

* [datashader.org](https://datashader.org)
* [holoviews.org](https://holoviews.org)
* [geoviews.org](https://geoviews.org)

While others are available as subdomains of holoviz.org:

* [hvplot.holoviz.org](https://hvplot.holoviz.org)
* [panel.holoviz.org](https://panel.holoviz.org)
* [lumen.holoviz.org](https://lumen.holoviz.org)
* [param.holoviz.org](https://param.holoviz.org)
* [colorcet.holoviz.org](https://colorcet.holoviz.org)

### Testing

There are four main kinds of tests that a HoloViz project may run:

* *Linters and Code formatter*: running programs that check Python source files for errors and styling issues. Most HoloViz projects rely on [Ruff](https://docs.astral.sh/ruff) in combination with [pre-commit](https://pre-commit.com/) to run the linters on every commit. You can run lint with `pixi run lint` and the `pre-commit` hook can be installed with `pre-commit run lint-install`.

> * *Unit tests*: the HoloViz projects rely on [pytest](https://docs.pytest.org/) to run their unit tests, sometimes with some additional pytest plugins. The `pixi run test-unit` will run these tests.

Reviewer: ```suggestion
* *Unit tests*: the HoloViz projects rely on [pytest](https://docs.pytest.org/) to run their unit tests, sometimes with some additional pytest plugins. The `pixi run test-unit` command will run these tests.
```

> * *Example tests*: the examples tests and the notebooks tests, in which the notebooks found in the `/examples` or `/doc` folder are all executed. Note that their output is not compared with any reference. Instead, the tests only fail if an error is raised while running the notebooks. The projects rely on pytest and [nbval](https://nbval.readthedocs.io). The `pixi run test-example` will run these tests.

Reviewer: I don't understand the first sentence.

Reviewer 2: It's saying what linters and code formatters are and what dependencies are used for it - `Ruff` and `pre-commit`

Reviewer: Sorry, I meant this one. I'm not sure what's the difference between examples tests and notebooks tests.

> *Example tests*: the examples tests and the notebooks tests, in which the notebooks found in the `/examples` or `/doc` folder are all executed.

The HoloViz project maintains a blog at https://blog.holoviz.org/ where new major releases are announced.

> The former PyViz-named blog is still alive at http://blog.pyviz.org/.

Reviewer: Just for reference, I was documenting that sort of thing here because there's a real connection between HoloViz and PyViz (https://github.com/pyviz/pyviz.org/issues/190). Now I opened this issue I think it's okay to remove this comment.

But yes, in general we should consider this document our knowledge base and not hesitate to add information even if they don't appear so relevant (it's better here than in someone's head only!)

Reviewer 2: OK. The comment is no longer there though.
